xml_handlers/testing_handler.php

```diff
@@ -235,6 +241,10 @@ public function endElement($parser, $name)
             $all_at_once = count($this->Builds) > 1;
             $parent_duration_set = false;
             foreach ($this->Builds as $subproject => $build) {
+                $build->StartTime = gmdate(FMT_DATETIME, $this->StartTimeStamp);
+                $build->EndTime = gmdate(FMT_DATETIME, $this->EndTimeStamp);
+                $build->UpdateBuild($build->Id, -1, -1);
```

Comment:
We both agree that it would be best if the XML parsers parsed all relevant data, and only saved it all at the very end.  Might be easier said than done.

Reply:
Investigate this possibility as a follow-up PR